app/models/skytap/template.rb

```
@@ -26,12 +28,18 @@ def self.pull
       api_templates.each do |api_template|
         next if api_template.get.publish_sets.blank?
 
-        template = Skytap::Template.with_course_type_scope.where(courses: { course_type: course_type }).find_or_create_by(skytap_id: api_template.id)
+        course_name, state_name = api_template.name.split('|').map(&:strip)
+
+        template = Skytap::Template.with_course_type_scope.where(courses: { course_type: course_type }).find_or_initialize_by(skytap_id: api_template.id)
+        template.state_name = state_name
+        template.save if template.changed?
+
         region = template.course_region.try(:region) || Region.find_or_create_by(name: api_template.region)
-        course = template.course_region.try(:course) || Course.find_or_create_by(name: api_template.name, course_type: course_type)
+        course = template.course_region.try(:course) || Course.find_or_create_by(name: course_name, course_type: course_type)
```

Consider `find_or_create_by!` on lines 37 & 38. `find_or_create_by` (no `!`) will fail silently but still return the unpersisted object if there are validation errors.